### DISCUSSION TITLE
how could I inject my server metrics into server-timing timingMiddleware?

### DISCUSSION THREAD

**Discussion by yashika** [nitrojs/nitro#284] [Q&A]
hello

may I ask how coud I inject some server statics into the server-timing metrics on nitro? maybe in the server middleware?

**Comment by TheAlexLichter** [comment]
My go to solution for now, based on https://github.com/unjs/nitro/commit/56c54ccda6f39849a5020006e27934367b53dfa1

```ts
export default defineEventHandler(async (event) => {
	globalThis.__timing__.logStart('someLogPoint')
    // Do something
	globalThis.__timing__.logEnd('someLogPoint')
})
```

To make it work with TS, I add global type for that:

```ts
// overrides.d.ts
declare global {
  var __timing__: {
    logStart: (id: string) => void
    logEnd: (id: string) => void
  }
}

export { }
```
